### ISSUE TITLE
I totally broke the Outline now...

### ISSUE THREAD

**Issue by lauragibbs** [elmsln/HAXcms#330]
Okay, I wanted to try to wrap things up so I could leave the site to sit for a while, so I blanked out the three problem pages that experienced out-of-whack problems: S000045, S000125, and S000145 (I think those are all the problem pages). First I manually fixed their title/location mismatches in developer mode of the outline view, and then I blanked out the content and removed those links from my records, reverting to my Google Doc content source. When I get back to using HAX again I'll fill in the content again for those pages. 

Then I went to delete the three "organizer" pages I had created to try to do some site management. These were R0001, R0001end, and Stories. Well, I deleted the Stories page, and that then blitzed the whole outline mode: everything that had appeared under Stories disappeared. The content is still there; it's just not accessible via the Outline. The Outline is basically all gone. Eeeeek!

So, the actual story pages are still in the system and show up... which is a relief! That means my existing links in Diigo do still work. For example, this link still works fine:
https://hax.lauragibbs.net/sites/story-folder/S000001

But that page has disappeared from the Outline, along with almost everything else; the only items left are a few Gullah stories that I had put under R0001, and thank goodness those didn't have their URLs changed. Presumably if I delete R0001, the Outline would go completely blank (?). But I'm not going to try that; I'm just going to leave as is.

You can see what the sidebar navigation / outline view looks like:
https://hax.lauragibbs.net/sites/story-folder/

There really are 130 pages that I created and which are in the system, but nowhere to be seen there in the Outline.

I'll keep prepping my content so that when there are some features to allow for some content organization, I can jump right back in! It just takes me a minute or two per page to load it up. So if/when HAX is ready for a site with hundreds of pages, I'm ready too! 

If it would help, I'm also glad to brainstorm about different approaches to organizing content online; I've used quite a few different systems, and they all have their advantages and disadvantages.

I'm also glad to do whatever kind of testing you want me to do to see how to recover from something like this. So if/when the Outline mode is working again, I can jump back in here and see if I can get that to work even if for my long-term purposes the Outline approach won't work. I'll just leave everything sit for now. :-)

**Comment by lauragibbs** [comment]
Screenshot of current situation in sidebar navigation, and in the Outline mode:

![Screen Shot 2019-07-17 at 6 26 40 PM](https://user-images.githubusercontent.com/19476543/61416095-95af6d80-a8c0-11e9-9661-767746a60417.png)
![Screen Shot 2019-07-17 at 6 27 16 PM](https://user-images.githubusercontent.com/19476543/61416104-9942f480-a8c0-11e9-8f23-f401ca80124e.png)

**Comment by btopro** [comment]
Yikes. So deleting folders doesn't delete the pages under it but it will basically leave them parent-less on the file system. If you hit the settings gear in the bottom right, there's a red button that pops up and says revert / roll back / something about commits. This is because every operation you send to the server is recorded in version control. So, you should be able to click that and it roll back a step and keep rolling it back steps until it works again. This is an aggressive action and it isn't broken out by outline vs content so if you changed content after this outline breaking operation it'll go away but it should be able to recover the outline.

**Comment by lauragibbs** [comment]
I figured I could un-do, but I left it there as an example of a behavior that is surely not expected, right? I'm not entirely sure how the outline is supposed to work, but I don't really want it to go back: I really do want to delete that Stories page. 
My guess is that you need to take some time at some point to figure out just what the outline is for and how it is going to work. I'll be glad to jump in and test it some more when the kinks are worked out: I've got a nice set of pages here ready to work with, and I'm glad to treat it as a sandbox to play with and test things so long as my URLs stay stable -- which, as I learned yesterday, is not necessarily the case with the Outline. Apparently sometimes the Outline structure changes URLs, but sometimes it doesn't (that inconsistency was really weird; I suspect it has to do with creating pages inside the Outline mode...?).
If I get time before I leave for Austin I'll try to write up some thoughts on why the Outline mode will not work for me as a way to manage site content, even if it might be a tool to manage the site navigation...? Site navigation is one thing, and site content management is another, and my guess is you are going to need to find two different ways to do that, using Outline as one option for managing site navigation, but something else more stable for site content...?
Anyway, just a quick thought; I'll try to write up some more thoughts from a content creation perspective before I leave for Austin! :-)

**Comment by btopro** [comment]
so I agree w/ the feedback that it needs work but I'm trying to provide a path forward that side steps it a bit
- roll back to it being in an outline yet they are at least there
- then if you open the outline and outdent the pages currently "in the folder" then hit save and they'll exist
- then you can delete the "folder"

Adding pages via the + button would satisfy adding pages.

I've pinged @nikkimk w/ this and other threads and it's something we'll be working on for Fall. Enjoy Austin and thanks for working through these sorts of things to help us grow!

**Comment by lauragibbs** [comment]
I resorted to adding the pages in the outline because adding pages with the + button was popping them into places unpredictably, not consistently. There are appx. 120 pages in that folder; I am not going to outdent them all manually in order to delete a single page. Probably I never should have tried indenting them to start with, but that was the only way I could make use of the outline mode to try to get some control over the site navigation. 
So, this is a good case study to use to think about: if you want to use a folder-based navigation/organization system (and I use PBWorks wiki software which has a folder-based approach, and that has worked great for me), then you need to build in some tools that will make that work: there needs to be a distinction between folders and pages, and you need to have a system to multi-select items in folders to move them from one folder to another, etc.
But is a folder-based system what you really want for navigation? What you really want for content management? The folder option has good sides, and bad sides.
Right now, though, what you've got is not exactly a folder option; the Outline mode is kind of like having folders, but also not like having folders. This problem with deleting a single page and losing the Outline is a symptom of that problem.
I will try to write up more detailed thoughts about this because it is super-important stuff, and we can treat my collection of pages here as a giant sandbox to play around with. Although it's not even that giant: 130 pages is not really a lot if you foresee HAX as supporting a course, for example, where hundreds of pages of content are easily the norm, especially if the course is textbook-free relying on web content.
For comparison, the UnTextbook I built for my Myth-Folklore class has 2700 pages; I built that in a single summer using Blogger, which is a great tool for creating content fast and keeping it super-organized (with labels), and it's worked for about 5 years now, no problems -- although I'd prefer to use a platform I control more fully than Blogger!
So, that is also something to think about: I had assumed that HAX could accommodate an ever-expanding number of pages. If that's not the case, I totally understand, and I can be a happy test-user for a small amount of content like this too! Small sites can be useful also.
For my own long-term purposes, though, I need something that can handle really large numbers of pages. That's why I rely on blog and wiki platforms; they are geared around the idea of lots of content with flexible, evolving navigation. And see also Flickr that I mentioned earlier; Flickr is an excellent example of how to manage massive quantities of content while letting users stay organized and create a really good navigation options for their users too. Yes, images are not pages, but if you think about a page as data and an image as data, it's not so different, at least from my perspective as a content creator anyway. :-)

**Comment by lauragibbs** [comment]
I wrote up some notes about large-ish sites and content management:
https://mytalespinner.blogspot.com/2019/07/day-76-some-quick-thoughts-about.html

I hope it is helpful! I'm thinking I need to create a different kind of HAX experiment for now instead of plunging into the beginning of a large site, something small so that I can keep playing with the system and waiting to see what kind of content management options will emerge for larger sites! :-)

**Comment by btopro** [comment]
@nikkimk --^ Link above.

Once Nikki gets her claws into HAXcms (she's been in complex elements like video-player and stepping into HAX form usability) then I know she'll make it crazy.

To your questions in the post, at a technical level, 1000s of pages are possible even in the current configuration. The UX of 1000s is the issue (as expressed) but not the system itself. I've loaded test sites with I believe 2300+ pages just to ensure the front end could handle it. Invoked about a 6 second slow down (vs like a 2 page site) but loaded and was still very responsive.

Definitely food for thought on the UX side (like the galleries notion).

Some past prototypes I've played with as far as "outline" helpers (short videos for more food for thought):
- https://www.youtube.com/watch?v=XDsuFrLWY5w
- https://www.youtube.com/watch?v=nE2cDT4z9IE

These show some ui patterns to make "outlining" a combination of hierarchy + card sorting + node relationship views. We never integrated this as it was for idea generation about process but might give some ideas as far as possibilities / what might make it easier to work with.

**Comment by lauragibbs** [comment]
OHHHHH, you made my day! I am so glad that a huge site is possible. I was not sure of technical limitations that might prevent that.
So....... count me in for a huge site full of stories!
But this Outline Mode is just killing me. My stuff is not hierarchical content that I can organize in outline mode. If/when there are some ways to organize stuff that it's not in a hierarchy like that, I will jump right back in.
It would even work for me just to have it be in alphabetical order by file name (I am super-strict about my filenames), so that I can use Diigo to do all the tagging.
That was my original plan, but it started to fall apart when there was all the weirdness in the index listing.
Anyway, I am really glad that there are not inherent limitations on numbers of pages. That means it's just a matter of finding ways to organize and work with lots of content. Yay!!!!!!!!!!!

**Comment by btopro** [comment]
I've met 3 other faculty like you that use the phrase "My content is not hierarchical". Flat disconnected data is a hierarchy in its lack of hierarchy :). Anyway, we'll be working on some of the outlining and haxcms specific UX/UI pieces this next month so tune back in in a few weeks and see if we've moved the ball forward on this at all. Thank you for being willing to bump into the rough edges of what we're doing. It helps us spread the word and move forward based on the needs of our user-base (as faculty are our primary audience as far as creation side of the house).

**Comment by michael-collins** [comment]
@lauragibbs I'm happy to work with you to wireframe up ideas you might have. If you see something particularly inspiring from an interface perspective, feel free to post it in #247 .

**Comment by lauragibbs** [comment]
Ha ha, I really mean it is not hierarchical. You can assign random numbers to my stories... just keep my URLs stable. :-)
I'm excited to see what takes shape........

**Comment by btopro** [comment]
largely solving this issue, need to do more battle testing via use-cases and improve a little more around the edges, BUT... after I was able to replicate this in a few sites causing me to dig deeper into the cause I went back and fixed this primarily on the front end.

English: I fixed the data issues in the outline + added a bunch of additional improvements that helps with managing larger groups of content, duplication of content, collapsing / toggling content, as well as (what broke the outline) fixing the cross browser issues that would cause pages to get lost.

Too little too late I know @lauragibbs but thank you for reporting this and giving me clues that led to figuring this out.
This is currently available for playing with on the haxtheweb.org website and will be released in the coming version after some more testing. Video too: https://www.youtube.com/watch?v=TrqEkMU5bBQ

![Uploading 2019-09-25_21-13-40.png…](haxtheweb.org)

Closing for now as I've been able to generate and move content around a lot without running into these issues (which I was able to previously generate).
